src/hooks/useTypeSafeContext.ts

```diff
@@ -0,0 +1,11 @@
+import { useContext } from "react";
+
+/**
+ * equivalent to `useContext` from react - but using this hook removes null check on every values from context.
+ * Use this one instead of the one from React
+ */
+export function useTypeSafeContext<TContext>(passedContext: React.Context<TContext | undefined>): TContext {
```

Comment:
❓  Use this component instead of `useContext` from react as the comment suggests. Otherwise you'll need to set `?` operator everywhere ❓